blockmanager.go

```diff
@@ -232,30 +245,30 @@ func (b *blockManager) findNextHeaderCheckpoint(height int32) *chaincfg.Checkpoi
 // download/sync the blockchain from.  When syncing is already running, it
 // simply returns.  It also examines the candidates for any which are no longer
 // candidates and removes them as needed.
-func (b *blockManager) startSync(peers *list.List) {
+func (b *blockManager) startSync() {
 	// Return now if we're already syncing.
 	if b.syncPeer != nil {
 		return
 	}
 
+	// Once the segwit soft-fork package has activated, we only
+	// want to sync from peers which are witness enabled to ensure
+	// that we fully validate all blockchain data.
+	segwitActive, err := b.chain.IsDeploymentActive(chaincfg.DeploymentSegwit)
+	if err != nil {
+		bmgrLog.Errorf("Unable to query for segwit soft-fork state: %v", err)
+		return
+	}
+
 	best := b.chain.BestSnapshot()
-	var bestPeer *serverPeer
-	var enext *list.Element
-	for e := peers.Front(); e != nil; e = enext {
-		enext = e.Next()
-		sp := e.Value.(*serverPeer)
-
-		// Once the segwit soft-fork package has activated, we only
-		// want to sync from peers which are witness enabled to ensure
-		// that we fully validate all blockchain data.
-		segwitActive, err := b.chain.IsDeploymentActive(chaincfg.DeploymentSegwit)
-		if err != nil {
-			bmgrLog.Errorf("Unable to query for segwit "+
-				"soft-fork state: %v", err)
+	var bestPeer *peerpkg.Peer
+	for peer, state := range b.peerStates {
+		if !state.syncCandidate {
 			continue
 		}
-		if segwitActive && !sp.IsWitnessEnabled() {
-			bmgrLog.Infof("peer %v not witness enabled, skipping", sp)
+
+		if segwitActive && !peer.IsWitnessEnabled() {
+			bmgrLog.Warnf("peer %v not witness enabled, skipping", peer)
```

Comment:
Looks like this got reverted in the shuffle.  Previous Comment:

This should be a Debugf. I know it was already an Infof, but I obviously missed that on the segwit review. There is no reason to notify users at an info level about this.

Since segwit isn't active this isn't being hit yet, but once it does, this could get quite noisy.